flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointStorageLoader.java

```diff
@@ -162,14 +163,19 @@ public static CheckpointStorage load(
         Preconditions.checkNotNull(classLoader, "classLoader");
         Preconditions.checkNotNull(configuredStateBackend, "statebackend");
 
-        if (configuredStateBackend instanceof CheckpointStorage) {
+        StateBackend rootStateBackend =
+                (configuredStateBackend instanceof ProxyStateBackend)
+                        ? ((ProxyStateBackend) configuredStateBackend).getProxiedStateBackend()
+                        : configuredStateBackend;
```

Comment:
This is the only place needing to unwrap for "state backend"; and it is because some of the backends are both state backend + checkpoint storage (for now).

There are some other places to check whether it is XXXKeyedStateBackend might be cofusing with this one.